### PR TITLE
feat(aiven_kafka_topic): validate conflict on plan

### PR DESCRIPTION
**Pull request by byashimov**
Validates topic conflicts on plan command.

### CODE REVIEW

**Comment by byashimov** [comment]
[tests](https://github.com/aiven/terraform-provider-aiven/actions/runs/8688594421)

**Comment by byashimov** [comment]
resolved conflicts

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go [hunk 1]
```diff
@@ -275,6 +278,25 @@ func ResourceKafkaTopic() *schema.Resource {
 				}
 			}
 
+			// Validates topic conflict for new topics
+			if d.Id() != "" {
```
Comment (Serpentiel) [inline internal/sdkprovider/service/kafkatopic/kafka_topic.go:282]: why not `d.IsNewResource()` instead?

**Review by Serpentiel — COMMENTED** [review]
lgtm, but there's a small question

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go (hunk 1 above)
Comment (byashimov) [reply internal/sdkprovider/service/kafkatopic/kafka_topic.go:282]: `schema.ResourceDiff` doesn't have such method.

**Review by Serpentiel — APPROVED** [review]

**Pull request merged**